scio-core/src/main/scala/com/spotify/scio/values/PairSCollectionFunctions.scala

```diff
@@ -353,6 +352,39 @@ class PairSCollectionFunctions[K, V](val self: SCollection[(K, V)])
     hotJoined ++ chillJoined
   }
 
+  /**
+   * Full outer join for cases when `this` is much larger than `that` which cannot fit in memory,
+   * but contains a mostly overlapping set of keys as `this`. A Bloom Filter of keys in `that` is
+   * used to split `this` into 2 partitions. Only those with keys in the filter go through the
+   * join and the rest are concatenated. This is useful for joining historical aggregates with
+   * incremental updates.
+   * @group join
+   * @param thatNumKeys estimated number of keys in `that`
+   * @param fpProb false positive probability when computing the overlap
+   */
+  def bloomFilterOuterJoin[W: ClassTag](that: SCollection[(K, W)],
```

Comment:
How about `sparseOuterJoin`, would be more consistent with `skewJoin`. `skew` is partitioning to get the hot keys, and `sparse` is partitioning to get the "cold" keys.